src/storage-types/src/connections/aws.rs

```diff
@@ -109,10 +110,11 @@ impl AwsCredentials {
         &self,
         connection_context: &ConnectionContext,
     ) -> Result<impl ProvideCredentials, anyhow::Error> {
-        let secrets_reader = connection_context.secrets_reader.as_ref();
+        let secrets_reader = &connection_context.secrets_reader;
         Ok(Credentials::from_keys(
             self.access_key_id
-                .get_string(secrets_reader)
+                // We will already be contained within a tokio task from `load_sdk_config`.
+                .get_string(InTask::No, secrets_reader)
```

Comment:
this feels like a leaky abstraction..
I think it would be cleaner if this also accepted an `in_task` parameter and `load_sdk_config` set it to `InTask::No` since it knows it's already in a task

Reply:
sounds good!